### PR TITLE
[DRAFT] Introduce file loader

### PR DESCRIPTION
**Pull request by mhs4670go**
This draft introduces file loader. 

Related : #2166
ONE-DCO-1.0-Signed-off-by: seongwoo <mhs4670go@naver.com>

### CODE REVIEW

**Comment by mhs4670go** [comment]
@seanshpark How about this change?

**Comment by seanshpark** [comment]
> @seanshpark How about this change?

Seems OK. @parjong , how about you?

**Comment by seanshpark** [comment]
@mhs4670go , can you prepare other projects like so?

And we should not forget to update SE process related files that has module names...

File: compiler/fileloader/include/fileloader/FileLoader.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    return _size;
+  }
+
+private:
+  const std::string _path;
+  DataBuffer _data;
+  std::streampos _size;
+};
+
+} // namespace fileloader
```
Comment (parjong) [inline compiler/fileloader/include/fileloader/FileLoader.h:74]: This name looks too general ;-)

File: compiler/fileloader/include/fileloader/FileLoader.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    // reserve capacity
+    _data.reserve(_size);
+
+    // read the data
+    file.read(_data.data(), _size);
+    if (file.fail())
+      return nullptr;
+
+    return _data.data();
```
Comment (parjong) [inline compiler/fileloader/include/fileloader/FileLoader.h:60]: IMHO This design is a bit dangerous. Any bug from out-of-bound read/write is very difficult to debug :(

File: compiler/fileloader/include/fileloader/FileLoader.h (hunk 1 above)
Comment (seanshpark) [reply compiler/fileloader/include/fileloader/FileLoader.h:74]: name it `filefox` ?
- fileloader + firefox -> filefox

Comment (mhs4670go) [reply compiler/fileloader/include/fileloader/FileLoader.h:74]: @parjong But it really does load any file..

Comment (parjong) [reply compiler/fileloader/include/fileloader/FileLoader.h]: > But it really does load any file..

This implementation is not the unique way to load a file ;-D

**Pull request closed without merging**